src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -354,22 +363,50 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             traceError(`Failed to create notebook controller for ${kernelConnection.id}`, ex);
         }
     }
-
+    private async setAsActiveControllerForTests(controller: VSCodeNotebookController, notebook: NotebookDocument) {
+        // Only when running tests should we force the selection of the kernel.
```

Comment:
Moved this from controller to controller manager.

Reply:
Good change, makes more sense here.